### DISCUSSION TITLE
Markdown Editor Box

### DISCUSSION THREAD

**Discussion by Pkcarreno** [CommunityToolkit/Labs-Windows#366] [Ideas]
Why is a wysiwyg component so complicated that it can support Markdown formatting in the same way as `RichEditBox`?

P.S: I am new to windows development and I would like to make a component like this since I can't get one, but I don't know where to start.

**Comment by michael-hawker** [comment]
Great question @Pkcarreno, the problem as to why this is a complicated component is that `RichEditBox` only knows how to deal with `RTF`. So you either need to recreate all the editor bits to make something that works with Markdown (which is also complicated) OR deal with converting between `RTF` and `Markdown` which is also complicated and there's no great available library that's freely available to do so.

It certainly would be a nice control, but our first priority is to re-work how our existing markdown viewer component works. See CommunityToolkit/WindowsCommunityToolkit#4663.

**Reply by michael-hawker** [reply]
PR for the MarkdownTextBlock replacement is now here: https://github.com/CommunityToolkit/Labs-Windows/pull/480
